### PR TITLE
RDX: Use image labels for the installed extensions list

### PR DESCRIPTION
**Pull request by mook-as**
- Expose the image labels on extensions
- Convert the front end to use API calls to get extension information, rather than IPC calls. (This involves changing the API to return information about the extensions beyond just the version.)
- In the UI in the installed extensions list, use the new labels (falling back to the extension ID) instead of the UI title.

This is meant to help with #4543 but we should double-check after merging if this solves all parts of it.

### CODE REVIEW

File: pkg/rancher-desktop/store/extensions.ts
Comment (rak-phillip) [inline pkg/rancher-desktop/store/extensions.ts:41]: We never reset `inError` to false if there is an error followed by a success. This will put us in a state where we can have both a collection of extensions and `inError: true`. 

We should always set `inError: false` if the fetch is successful.

File: pkg/rancher-desktop/main/extensions/extensions.ts
Comment (rak-phillip) [inline pkg/rancher-desktop/main/extensions/extensions.ts:124]: Should we be wrapping any of this in a `try...catch` statement to handle any errors involved with reading the file or parsing `stdout`?

File: pkg/rancher-desktop/layouts/default.vue
Comment (rak-phillip) [inline pkg/rancher-desktop/layouts/default.vue:86]: This ends up being quite elegant ✨

**Review by rak-phillip — CHANGES_REQUESTED** [review]
This functions well and looks great. I left you my feedback while reviewing the changes.

**Review by mook-as — COMMENTED** [review]
Comments addressed.

File: pkg/rancher-desktop/pages/extensions/installed.vue
Comment (rak-phillip) [reply pkg/rancher-desktop/pages/extensions/installed.vue]: Thanks, moving this into a method definitely helps to keep the template simple.

File: pkg/rancher-desktop/store/credentials.ts
Comment (rak-phillip) [reply pkg/rancher-desktop/store/credentials.ts:16]: Right, I don't like having this attached to the credentials store, but this might be the best place for it for now.. I think that we leave it as is for now and give this more thought if we find ourselves doing more changes like this. 

I'm fine with this approach just as long as we take care to keep its surface area restricted to store to store communication.

File: pkg/rancher-desktop/store/extensions.ts
Comment (rak-phillip) [reply pkg/rancher-desktop/store/extensions.ts:41]: Out of band discussion: `inError` isn't used anywhere for this feature. We will remove from this PR and add again later if we decide to do anything with the error state.

**Review by rak-phillip — APPROVED** [review]
LGTM - waiting for CI before merging

**Pull request merged**
